aggregation/src/main/java/fk/prof/aggregation/serialize/Serializer.java

```diff
@@ -0,0 +1,47 @@
+package fk.prof.aggregation.serialize;
+
+import com.google.protobuf.Message;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.zip.Adler32;
+
+/**
+ * Base class for serializers.
+ * Provided helper methods to serialize Integer and protobuf message.
+ * Protobuf message will be serialized in the format [size][message][checksum].
+ * @author gaurav.ashok
+ */
+public abstract class Serializer<T> {
+
+    public abstract void serialize(T object, OutputStream os) throws IOException;
+
+    public static void writeInt32(int value, OutputStream os) throws IOException {
+        byte[] bytes = {(byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)value};
+        os.write(bytes);
+    }
+
+    /**
+     * Write protobuf message with its size followed by a checksum.
```

Comment:
Well the exact thought came to mind when I was putting that comment. But then I thought, "hmm, since I am not explicitly mentioning where the size goes (before or after) maybe it makes sense". So in result I ended up putting a format and then named the method writeFenced (meaning the data is actually surrounded by its size and checksum).